form/src/main/java/org/dhis2/form/ui/provider/FieldProvider.kt

```diff
@@ -87,39 +154,317 @@ private fun ProvideInputText(
             fieldUiModel.focused -> InputShellState.FOCUSED
             else -> InputShellState.UNFOCUSED
         },
-        supportingText = mutableListOf<SupportingTextData>().apply {
-            fieldUiModel.error?.let {
-                add(
-                    SupportingTextData(
-                        it,
-                        SupportingTextState.ERROR,
-                    ),
-                )
-            }
-            fieldUiModel.warning?.let {
-                add(
-                    SupportingTextData(
-                        it,
-                        SupportingTextState.WARNING,
-                    ),
-                )
-            }
-            fieldUiModel.description?.let {
-                add(
-                    SupportingTextData(
-                        it,
-                        SupportingTextState.DEFAULT,
-                    ),
-                )
-            }
+        supportingText = supportingText,
+        legendData = fieldUiModel.legend?.let {
+            LegendData(Color(it.color), it.label ?: "", null)
+        },
+        inputText = value ?: "",
+        onValueChanged = {
+            value = it
+            intentHandler(
+                FormIntent.OnSave(
+                    fieldUiModel.uid,
+                    value,
+                    fieldUiModel.valueType,
+                ),
+            )
+        },
+    )
+}
+
+@Composable
+private fun ProvideIntegerPositive(
+    fieldUiModel: FieldUiModel,
+    intentHandler: (FormIntent) -> Unit,
+) {
+    var value by remember {
+        mutableStateOf(fieldUiModel.value)
+    }
+
+    val supportingText by remember(fieldUiModel) {
```

Comment:
I think that we don't need to remember supporting test, if we made it stateless in the implementation will be updated everytime it is updated. What do you think?